civicase.php

```diff
@@ -30,10 +30,19 @@ function civicase_civicrm_tabset($tabsetName, &$tabs, $context) {
           ]);
         }
         if ($tab['id'] === 'activity') {
+          $activity_types = array_flip(CRM_Activity_BAO_Activity::buildOptions('activity_type_id', 'validate'));
           $useAng = TRUE;
           $tab['url'] = CRM_Utils_System::url('civicrm/case/contact-act-tab', [
             'cid' => $context['contact_id'],
           ]);
+          // Exclude bulk email activity type from the Activity count because
+          // there are issues with target contact for this activity type.
+          // To remove this code once issue is fixed from core.
+          $params = [
+            'activity_type_exclude_id' => $activity_types['Bulk Email'],
```

Comment:
@tunbola cant we put directly
`'activity_type_id' => ['!=' => 'Bulk Email'],`
this way we dont need to fetch the Activity IDs.

Reply:
@deb1990 Please note that the function is not expecting the format for parameters for an API, rather it prepares it's own parameters before it is sent to the Activity.Get API. Also note that the `activity_type_exclude_id` parameter is not a valid parameter for the API, it is only needed by this function to do some processing before it calls the API.

Please see: https://github.com/civicrm/civicrm-core/blob/afe8045c997cf28ee7d4b9d5e1eef799a76b2e84/CRM/Activity/BAO/Activity.php#L2438
https://github.com/civicrm/civicrm-core/blob/afe8045c997cf28ee7d4b9d5e1eef799a76b2e84/CRM/Activity/BAO/Activity.php#L883
To see what the function is expecting when we need to exclude an activity ID.